dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/authorization/groups/impl/saml/SamlGroupService.java

```diff
@@ -0,0 +1,92 @@
+package edu.harvard.iq.dataverse.authorization.groups.impl.saml;
+
+import edu.harvard.iq.dataverse.RoleAssigneeServiceBean;
+import edu.harvard.iq.dataverse.actionlogging.ActionLogServiceBean;
+import edu.harvard.iq.dataverse.persistence.ActionLogRecord;
+import edu.harvard.iq.dataverse.persistence.group.SamlGroup;
+import edu.harvard.iq.dataverse.persistence.group.SamlGroupRepository;
+import edu.harvard.iq.dataverse.persistence.user.AuthenticatedUser;
+import edu.harvard.iq.dataverse.persistence.user.RoleAssignment;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.ejb.Stateless;
+import javax.inject.Inject;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+@Stateless
+public class SamlGroupService {
+    private static final Logger logger = LoggerFactory.getLogger(SamlGroupService.class);
+
+    private RoleAssigneeServiceBean roleAssigneeSvc;
+
+    private ActionLogServiceBean actionLogSvc;
+
+    private SamlGroupRepository repository;
+
+    // -------------------- CONSTRUCTORS --------------------
+
+    public SamlGroupService() { }
+
+    @Inject
+    public SamlGroupService(RoleAssigneeServiceBean roleAssigneeSvc, ActionLogServiceBean actionLogSvc, SamlGroupRepository repository) {
+        this.roleAssigneeSvc = roleAssigneeSvc;
+        this.actionLogSvc = actionLogSvc;
+        this.repository = repository;
+    }
+
+    // -------------------- LOGIC --------------------
+
+    public List<SamlGroup> findAll() {
+        return repository.findAll();
+    }
+
+    public SamlGroup findById(Long id) {
+        return repository.findById(id).orElse(null);
+    }
+
+    public SamlGroup save(String name, String entityId) {
+        ActionLogRecord alr = new ActionLogRecord(ActionLogRecord.ActionType.GlobalGroups, "shibCreate");
+        alr.setInfo(name + ": " + entityId);
+
+        SamlGroup saved = repository.saveAndFlush(new SamlGroup(name, entityId));
+        actionLogSvc.log(alr);
+        return saved;
+    }
+
+    public Set<SamlGroup> findFor(AuthenticatedUser authenticatedUser) {
+        Set<SamlGroup> groupsForUser = new HashSet<>();
+        String samlIdPEntityId = authenticatedUser.getSamlIdPEntityId();
+        logger.debug("IdP for user " + authenticatedUser.getIdentifier() + " is " + samlIdPEntityId);
+        if (samlIdPEntityId != null) {
+            groupsForUser.addAll(repository.findByEntityId(samlIdPEntityId));
+        }
+        return groupsForUser;
+    }
+
+    public boolean delete(SamlGroup doomed) throws Exception {
```

Comment:
Method can't return false, so I think that we could omit return type (we communicate fails by Exception throwing)